docs/diagrams/from-comm.puml

```diff
@@ -0,0 +1,114 @@
+
+@startuml
+title ID-contact communication-initiated authentication v0.3.0
+
+actor Citizen as citizen
+participant "Core widget" as widget
+participant "ID Contact Core" as core
+participant "Auth plugin" as auth_plugin
+participant "Auth widget" as auth_widget
+participant "Comm plugin" as comm_plugin
+participant "Comm widget" as comm_widget
+actor Agent as agent
+
+== Initialize session ==
+
+activate citizen
+citizen->widget: navigate
+activate widget
+widget-->citizen: page
+citizen->widget: click
+widget->core: sessionOptions(purpose)
+activate core
+core->core: getAuthMethods(purpose)
+core->core: getCommMethods(purpose)
+core-->widget: ok(commMethods, authMethods),
+deactivate core
+widget-->citizen: commMethods
+
+== Choose communication method ==
+
+citizen->widget: selectCommMethod
+citizen->widget: start
+widget->core: start(commMethod, purpose)
+activate core
+core->comm_plugin: start(purpose, noAuth=true)
+activate comm_plugin
+comm_plugin-->core: ok(commUrl)
+core-->widget: redirect(commUrl)
+deactivate core
+widget-->citizen: redirect(commUrl)
+deactivate widget
+
+== Start conversation ==
+
+citizen->comm_widget: load(commUrl)
+activate comm_widget
+comm_widget-->citizen: page
+
+citizen->comm_widget: startConversation
+comm_widget->comm_plugin: startConversation
+
+comm_plugin->comm_plugin: assignAgent(purpose)
+comm_plugin->comm_plugin: createConversation(agentId)
+
+== Conversation ongoing ==
+comm_plugin->agent: newConversation
+
+activate agent
+citizen<->agent: conversation
+
+== Authentication during conversation ==
+agent->comm_plugin: requestAuth
+comm_plugin->comm_widget: startAuth(purpose, attrUrl, commUrl)
+comm_widget->widget: startAuth(purpose, attrUrl, commUrl)
+activate widget
```

Comment:
Is the widget loaded within the UI of the comm plugin er is the citizen being redirected? The former would be nice, but it would mean that the comm plugin needs to present the auth options available.

Reply:
For now, i would suggest we focus on having the widget open separately (which is what this flow mostly suggest). Longer term we could look at options for the communication widget having a more active role, either embedding or directly providing the choice widget.